libs/ui/src/themes/make_theme.ts

```diff
@@ -96,47 +97,60 @@ const colorThemes: Record<ColorMode, ColorTheme> = {
 };
 
 const INCHES_PER_MM = 1 / 25.4;
-const PIXELS_PER_INCH = 72;
 
-function mmToPx(mm: number): number {
-  // TODO: This is just a stop-gap proxy - need to add device detection logic to
-  // determine what screen we're on (ELO/VSAP/laptop) and use pre-defined
-  // ratios, since we can measure physical display dimensions more accurately.
-  const { devicePixelRatio } = window;
+/** Standard resolution for web images/measurements: */
+const PIXELS_PER_INCH_WEB = 72;
 
-  return mm * INCHES_PER_MM * devicePixelRatio * PIXELS_PER_INCH;
+const SCREEN_WIDTH_PIXELS_ELO_15 = 1080;
+const SCREEN_WIDTH_INCHES_ELO_15 = 7.62;
+
+const SCREEN_WIDTH_PIXELS_ELO_13 = 1080;
+const SCREEN_WIDTH_INCHES_ELO_13 = 6.51;
+
+interface SizeThemeParams {
+  screenType: ScreenType;
+  sizeMode: SizeMode;
 }
 
-function getFontSize(mode: SizeMode): number {
+/** PPI calculation functions by screen type: */
+const devicePixelsPerInch: Record<ScreenType, () => number> = {
+  builtIn: () => window.devicePixelRatio * PIXELS_PER_INCH_WEB,
+  elo13: () => SCREEN_WIDTH_PIXELS_ELO_13 / SCREEN_WIDTH_INCHES_ELO_13,
+  elo15: () => SCREEN_WIDTH_PIXELS_ELO_15 / SCREEN_WIDTH_INCHES_ELO_15,
+};
+
+function mmToPx(mm: number, screenType: ScreenType): number {
+  const pixelsPerInch: number = devicePixelsPerInch[screenType]();
+
+  return mm * INCHES_PER_MM * pixelsPerInch;
+}
+
+function getFontSize({ screenType, sizeMode }: SizeThemeParams): number {
   // Use the average midpoint value of the relevant VVSG size range.
   const capitalLetterHeightMm =
-    (VVSG_CAPITAL_LETTER_HEIGHTS_MM[mode].min +
-      VVSG_CAPITAL_LETTER_HEIGHTS_MM[mode].max) /
+    (VVSG_CAPITAL_LETTER_HEIGHTS_MM[sizeMode].min +
+      VVSG_CAPITAL_LETTER_HEIGHTS_MM[sizeMode].max) /
     2;
 
   const fullFontHeightMm =
     capitalLetterHeightMm * CAPITAL_HEIGHT_TO_FULL_FONT_HEIGHT_RATIO;
 
-  return mmToPx(fullFontHeightMm);
+  return mmToPx(fullFontHeightMm, screenType);
 }
 
 const VVSG_MIN_TOUCH_AREA_SIZE_MM = 12.7;
-const VVSG_MIN_TOUCH_AREA_SIZE_PX = mmToPx(VVSG_MIN_TOUCH_AREA_SIZE_MM);
 
 const VVSG_MIN_TOUCH_AREA_SEPARATION_MM = 2.54;
-const VVSG_MIN_TOUCH_AREA_SEPARATION_PX = mmToPx(
-  VVSG_MIN_TOUCH_AREA_SEPARATION_MM
-);
 
-const sizeThemes: Record<SizeMode, SizeTheme> = {
-  s: {
+const sizeThemes: Record<SizeMode, (p: SizeThemeParams) => SizeTheme> = {
```

Comment:
Out of curiosity, how is this used? Are there references that need to be updated?

Reply:
Currently only used locally down at L316 to compute the size theme based on the the selected mode.
Considering refactoring this a bit, honestly - it's grown over time and getting a little cumbersome